src/server_manager/model/server.ts

```diff
@@ -107,6 +107,10 @@ export interface ManualServer extends Server {
 // Managed servers are servers created by the Outline Manager through our
 // "magic" user experience, e.g. DigitalOcean.
 export interface ManagedServer extends Server {
+  // Indicates how far installation has progress.
+  installProgress(): number;
+  // Enables notifications related to installation progress.
+  onInstallProgressChange: (progress: number) => void;
   // Returns a promise that fulfills once installation is complete.
   waitOnInstall(): Promise<void>;
```

Comment:
You can delete waitOnInstall() and isInstallCompleted(). See below

Reply:
Done.  Note that this results in a single-frame "flash" of the progress screen when loading the Outline Manager with a DigitalOcean server already available, because checking for installation status is now asynchronous (but extremely fast).